field/TinyLang/Field/Generator.hs

```diff
@@ -75,77 +89,53 @@ chooseUniVar = elements . uniVars
    then sets the current free unique so that the Supply monad is in a
    state where any new calls to freshUnique will generate uniques that
    we haven't already used.  You then have to call
-   adjustUniquesForVars before running the generator (see
-   boundedArbitraryExpr).
+   adjustUniquesForVars before running the generator.
 
    Presumably to do this properly we'd need to do
    all name generation in the Supply monad, but then I don't think
    we'd be able to get our Arbitrary instances to have the right types
-   (???).  Alternatively, Var could provide a function to set the
-   state of the Unique supply.
-
+   (???).
 -}
```

Comment:
We can probably delete the last paragraph of this comment.  I added that when I was trying to work out the best way to get new uniques inside generators.

Reply:
I'd like to keep it to think about this later.